src/controllers/engine/controllerengine.cpp

```diff
@@ -995,7 +950,11 @@ void ControllerEngine::timerEvent(QTimerEvent *event) {
     }
 
     if (timerTarget.callback.isString()) {
-        internalExecute(timerTarget.callback.toString());
+        QJSValue callback = evaluateCodeString(timerTarget.callback.toString());
```

Comment:
Re-evaluating the code string repeatedly is inefficient. However, it hasn't practically been reported to be a problem, and the wiki now advises script developers to pass functions to `engine.beginTimer`... so I guess it's okay to leave it as is.

Reply:
Let's just evaluate the function when creating the timer.